encodings/sparse/src/lib.rs

```diff
@@ -164,13 +164,16 @@ impl SparseArray {
 
         let fill_array = ConstantArray::new(fill.clone(), array.len()).into_array();
         let non_top_mask = Mask::from_buffer(
-            compare(array, &fill_array, Operator::NotEq)?
-                .to_bool()?
-                .boolean_buffer()
-                .clone(),
+            fill_null(
+                &compare(array, &fill_array, Operator::NotEq)?,
+                Scalar::bool(true, Nullability::NonNullable),
+            )?
```

Comment:
I had fixed this before and when I extracted the sparsearray encode logic I lost it